discord/interactions.py

```diff
@@ -742,16 +749,45 @@ async def edit_message(
         if view is not MISSING:
             state.prevent_view_updates_for(message_id)
             payload["components"] = [] if view is None else view.to_components()
+
+        if file is not MISSING and files is not MISSING:
+            raise InvalidArgument("cannot pass both file and files parameter to edit_message()")
+
+        if file is not MISSING:
+            if not isinstance(file, File):
+                raise InvalidArgument("file parameter must be a File")
+            else:
+                files = [file]
+                if "attachments" not in payload:
+                    # we keep previous attachments when adding a new file
+                    payload["attachments"] = [a.to_dict() for a in msg.attachments]
+
+        if files is not MISSING:
+            if len(files) > 10:
+                raise InvalidArgument("files parameter must be a list of up to 10 elements")
+            elif not all(isinstance(file, File) for file in files):
+                raise InvalidArgument("files parameter must be a list of File")
+            else:
+                if "attachments" not in payload:
+                    # we keep previous attachments when adding new files
+                    payload["attachments"] = [a.to_dict() for a in msg.attachments]
```

Comment:
```suggestion
            if "attachments" not in payload:
                # we keep previous attachments when adding new files
                payload["attachments"] = [a.to_dict() for a in msg.attachments]
```

Remove unnecessary else branch after guard condition.